diff --git a/openthread b/openthread
index ee2049f..ada7bad 160000
--- a/openthread
+++ b/openthread
@@ -1 +1 @@
-Subproject commit ee2049fb48713b015c502f28ffcd464acd800de6
+Subproject commit ada7bada48d4e3b5e180d73936befac77a4fecba
diff --git a/src/radio.c b/src/radio.c
index 2869914..290d148 100644
--- a/src/radio.c
+++ b/src/radio.c
@@ -1795,7 +1795,7 @@ otError otPlatRadioClearSrcMatchExtEntry(otInstance *aInstance, const otExtAddre
     otError  error      = OT_ERROR_NONE;
     uint64_t local_addr = *((uint64_t *)aExtAddress); // ensures alignment
     uint8_t  idx;
-    otEXPECT_ACTION(idx   = rfCoreFindExtSrcMatchIdx(local_addr) != PLATFORM_RADIO_SRC_MATCH_NONE,
+    otEXPECT_ACTION((idx   = rfCoreFindExtSrcMatchIdx(local_addr)) != PLATFORM_RADIO_SRC_MATCH_NONE,
                     error = OT_ERROR_NO_ADDRESS);
 
     /* verify that we have a running or backgrounded rx command */
